x-pack/plugins/asset_manager/server/lib/asset_accessor.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Asset } from '../../common/types_api';
+import { AssetManagerConfig } from '../types';
+import { OptionsWithInjectedValues } from './accessors';
+import { GetHostsOptions } from './accessors/hosts';
+import { getHostsByAssets } from './accessors/hosts/get_hosts_by_assets';
+import { getHostsBySignals } from './accessors/hosts/get_hosts_by_signals';
+
+interface AssetAccessorClassOptions {
+  sourceIndices: AssetManagerConfig['sourceIndices'];
+  source: AssetManagerConfig['lockedSource'];
```

Comment:
I missed the review window for the last PR, but what's the thinking behind the config name `lockedSource`?

Reply:
The switch between reading from signals vs reading from assets.

Reply:
I called it that bc I anticipate wanting to allow the source to be fluid and adapt to available data and this config "locks it" to one or the other, but I didn't think that hard about it tbh 🤣